src/state_machine.rs

```diff
@@ -193,49 +193,58 @@ impl State {
 
     pub fn get_timed_out_tokens(&mut self) -> Vec<u64> {
         match *self {
-            State::Node(ref mut state) => state.get_timed_out_tokens(),
+            State::Bootstrapping(_) | State::Terminated => vec![],
             State::Client(ref mut state) => state.get_timed_out_tokens(),
```

Comment:
No because all the `state`s are actually different types. ([playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6249b0de048643e437f73096f10909c0))